### PR TITLE
Reorganize docs and fix some small bugs

### PR DESCRIPTION
**Pull request by cgokmen**
This PR:

* Moves some docs pages around
* Fixes all broken links in docs
* Replaces .html links with .md links as expected by mkdocs (the former works but mkdocs doesnt recognize them as cross links)
* Fixes config issue that makes griffe misunderstand returntypes
* Fixes typo in point goal reward potential
* Adds assertion to safeguard against uninitialized objects being saved
* Sets the default observation mode of robots to be rgb+proprio instead of the whole mess (bounding boxes are REALLY slow)
* Fixes a bug in ControllableObjectViewAPI that causes error outputs at load time

### CODE REVIEW

File: docs/miscellaneous/roadmap.md
Comment (cremebrule) [inline docs/miscellaneous/roadmap.md:1]: is this intentionally empty? placeholder for the future?

File: docs/modules/under_the_hood.md [hunk 1]
```diff
@@ -5,12 +5,6 @@ icon: material/car-wrench
 # Under the Hood: Isaac Sim Details
 In this page, we discuss the particulars of certain Isaac Sim features and behaviors.
 
-## Playing and Stopping
-TODO
```
Comment (cremebrule) [inline docs/modules/under_the_hood.md:9]: do we have a ticket to track these?

File: docs/modules/under_the_hood.md [hunk 2]
```diff
@@ -42,6 +36,28 @@ To conclude, with ENABLE_FLATCACHE enabled, there will be three concurrent repre
...
+
+There are certain operations that may only be performed when simulation is playing, e.g. when a PhysX representation is present. This include using Tensor APIs, getting physical state like collisions, impulses, etc., and using the physics simulation APIs. When the simulation is stopped, these operations will not be available.
+
+On the other hand, while operations like adding and removing objects are in theory supported by PhysX (e.g. they are supposed to be OK to call even when the simulation is playing), we have found that adding and removing objects may result in partially updated PhysX states and cause crashes. As a result, if you are running into issues with adding and removing objects, it is recommended to stop the simulation before doing so. When you play again, the scene will be reloaded from USD and the PhysX state will be recreated from scratch, leaving no trace of the addition/removal.
+
+## CPU and GPU dynamics and pipelines
+Isaac Sim (and thus OmniGibson) supports CPU and GPU dynamics, and when using GPU dynamics, it supports CPU and GPU pipelines. Here we discuss the differences between these and how they are used in OmniGibson.
```
Comment (cremebrule) [inline docs/modules/under_the_hood.md:53]: should we mention that we're in the process of fully deprecating CPU dynamics? (or is that no longer on our roadmap?)

File: mkdocs.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-yaml-language-server: $schema=https://squidfunk.github.io/mkdocs-material/schema.json
+# yaml-language-server: $schema=https://squidfunk.github.io/mkdocs-material/schema.json
```
Comment (cremebrule) [inline mkdocs.yml:1]: not sure what this does, just rubber stamping for now

File: mkdocs.yml [hunk 4]
```diff
@@ -74,6 +77,9 @@ plugins:
       handlers:
         python:
           paths: [omnigibson]
+          options:
+            docstring_options:
+              returns_named_value: false
```
Comment (cremebrule) [inline mkdocs.yml:82]: BLESS

File: omnigibson/object_states/object_state_base.py [hunk 5]
```diff
@@ -198,7 +198,7 @@ def update_cache(self, get_value_args):
     def cache_info(self, get_value_args):
         """
         Helper function to cache relevant information at the current timestep.
-        Stores it under @self._cache[<KEY>]["info"]
+        Stores it under @self._cache [<KEY>]["info"]
```
Comment (cremebrule) [inline omnigibson/object_states/object_state_base.py:201]: what's the point of this space?

File: omnigibson/prims/prim_base.py [hunk 6]
```diff
@@ -158,6 +164,22 @@ def state_size(self):
...
+            serialized (bool): If True, will return the state of this object as a 1D numpy array. Otherewise, will return
+                a (potentially nested) dictionary of states for this object
+
+        Returns:
+            dict or n-array: Either:
+                - Keyword-mapped states of this object, or
+                - encoded + serialized, 1D numerical np.array capturing this object's state
+        """
+        assert self._initialized, "Prim must be initialized before dumping state!"
+        return self._dump_state(serialized=serialized)
```
Comment (cremebrule) [inline omnigibson/prims/prim_base.py:181]: this should call super dump_state(), not the internal call _dump_state()

**Review by cremebrule — COMMENTED** [review]
overall LGTM, a few minor comments and one bug

File: docs/modules/under_the_hood.md (hunk 2 above)
Comment (cgokmen) [reply docs/modules/under_the_hood.md:53]: not sure if it is our roadmap rn so leaving this as is

File: mkdocs.yml (hunk 3 above)
Comment (cgokmen) [reply mkdocs.yml:1]: this essentially tells vscode what schema it should expect the file to match. the yaml language server in vscode knows to parse this directive even when it's commented out. but when it's not commented out, it gets passed as a config param into mkdocs too which gets confused and outputs warnings - which is why it needs to be commented

File: omnigibson/object_states/object_state_base.py (hunk 5 above)
Comment (cgokmen) [reply omnigibson/object_states/object_state_base.py:201]: the @ cross reference does not work with the space

Comment (cgokmen) [reply omnigibson/object_states/object_state_base.py:201]: and it outputs an error

File: docs/miscellaneous/roadmap.md
Comment (cgokmen) [reply docs/miscellaneous/roadmap.md:1]: yeah - hopefully can be actually written soon haha

File: omnigibson/prims/prim_base.py (hunk 6 above)
Comment (cgokmen) [reply omnigibson/prims/prim_base.py]: omg yes good catch. fixed

File: omnigibson/prims/prim_base.py [hunk 7]
```diff
@@ -178,7 +178,7 @@ def dump_state(self, serialized=False):
                 - encoded + serialized, 1D numerical np.array capturing this object's state
         """
         assert self._initialized, "Prim must be initialized before dumping state!"
-        return self._dump_state(serialized=serialized)
+        return self.dump_state(serialized=serialized)
```
Comment (cremebrule) [inline omnigibson/prims/prim_base.py:181]: this should be `super`, not `self`

File: mkdocs.yml (hunk 3 above)
Comment (cremebrule) [reply mkdocs.yml:1]: Can we add a comment above it specifying this?

File: omnigibson/prims/prim_base.py (hunk 7 above)
Comment (cgokmen) [reply omnigibson/prims/prim_base.py:181]: Fixed. Sorry for the double whammy here.

File: mkdocs.yml (hunk 3 above)
Comment (cgokmen) [reply mkdocs.yml:1]: Done.

**Pull request merged**